pkg/solana/logpoller/loader.go

```diff
@@ -475,34 +469,44 @@ func (p *orderedParser) sendReadySlots() error {
 			return errs
 		}
 
-		p.ready = remove(p.ready, rIdx)
+		p.ready = slices.Delete(p.ready, rIdx, rIdx+1)
 		rmvIdx = append(rmvIdx, idx)
 
 		delete(p.expect, block)
 		delete(p.actual, block)
 	}
 
 	for count, idx := range rmvIdx {
-		p.blocks = remove(p.blocks, idx-count)
+		p.blocks.Delete(idx - count)
 	}
 
 	return nil
 }
 
-func getIdx[T any](slice []T, match T) (int, bool) {
-	for idx, value := range slice {
-		if reflect.DeepEqual(value, match) {
-			return idx, true
-		}
-	}
+var (
+	errExpectationsNotSet = errors.New("expectations not set")
+)
+
+type blockHeap []uint64
 
-	return -1, false
+func (h blockHeap) Len() int           { return len(h) }
+func (h blockHeap) Less(i, j int) bool { return h[i] < h[j] }
+func (h blockHeap) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }
+
+func (h *blockHeap) Push(x any) {
+	*h = append(*h, x.(uint64))
 }
 
-func remove[T any](slice []T, s int) []T {
-	return append(slice[:s], slice[s+1:]...)
+func (h *blockHeap) Pop() any {
+	old := *h
+	n := len(old)
+	x := old[n-1]
+	*h = old[0 : n-1]
+
+	return x
 }
 
-var (
-	errExpectationsNotSet = errors.New("expectations not set")
-)
+func (h *blockHeap) Delete(idx int) {
+	old := *h
+	*h = slices.Delete(old, idx, idx+1)
+}
```

Comment:
I don't think you can use `slices.Delete` with a heap, because it has to do do some fancy rebalancing while removing an element to preserve the heap invariant.

I think we just need to call:
```
heap.Remove(h, idx)
```
and then it does it all for you. No need to define a custom `Delete` method

Reply:
Update: you shouldn't need either this or `heap.Remove`, just `heap.Pop`--see other comments